apps/client/src/services/initialize-socket.ts

```diff
@@ -1,11 +1,45 @@
+import {
+  PROTO_AUTH_TOKEN_TRANSFER,
+  PROTO_JOIN_LOBBY_REQUEST,
+} from '@razor/constants';
+import { AuthToken, InitialClientData } from '@razor/models';
 import { io, Socket } from 'socket.io-client';
 
 const SOCKET_ENDPOINT =
   process.env['NX_SOCKET_ENDPOINT'] || 'http://localhost:3000';
-const socket: Socket = io(SOCKET_ENDPOINT, {
+let authToken = '';
+
+const socket = io(SOCKET_ENDPOINT, {
+  auth: {
+    token: authToken,
+  },
+  autoConnect: false,
   withCredentials: true,
-});
+}) as Socket & { auth: { token: AuthToken } };
```

Comment:
Shall we separate this type delerations

Reply:
This is fixed in commit 4564f6717a440f69a2531402a290b8ef92b21055 under PR #73 
https://github.com/the-ai-team/razor/pull/73/files#diff-e7af1152c8e7f0a3b83f78d4ab66c7a8368764cc1b31f300ac8c5e8569602c1b

FYI: `initialize-socket` was renamed as `socket-communication` in same PR.